mbtiles/src/mbtiles.rs

```diff
@@ -223,7 +303,108 @@ impl Mbtiles {
             self.check_agg_tiles_hashes(&mut conn).await
         }
     }
+    pub async fn statistics<T>(&self, conn: &mut T) -> MbtResult<Statistics>
+    where
+        for<'e> &'e mut T: SqliteExecutor<'e>,
+    {
+        let file_size = query!(
+            "SELECT page_count * page_size as file_size FROM pragma_page_count(), pragma_page_size();"
+        ).fetch_one(&mut *conn)
+        .await?
+        .file_size
+        .expect("The file size of the MBTiles file shouldn't be None") as u64;
+
+        let page_size = query!("PRAGMA page_size;")
+            .fetch_one(&mut *conn)
+            .await?
+            .page_size
+            .unwrap() as u64;
+        let tile_infos_query = query!(
+            r#"SELECT
+                zoom_level AS zoom,
+                count( ) AS count,
+                min( length( tile_data ) ) AS smallest,
+                max( length( tile_data ) ) AS largest,
+                avg( length( tile_data ) ) AS average,
+                min(tile_column) as min_tile_x,
+                min(tile_row) as min_tile_y,
+                max(tile_column) as max_tile_x,
+                max(tile_row) as max_tile_y 
+            FROM tiles
+            GROUP BY zoom_level"#
+        );
```

Comment:
i reformatted it a bit to fit with all other SQL statement formattings:

```suggestion
        let tile_infos_query = query!(
            "
    SELECT zoom_level             AS zoom,
           count()                AS count,
           min(length(tile_data)) AS smallest,
           max(length(tile_data)) AS largest,
           avg(length(tile_data)) AS average,
           min(tile_column)       AS min_tile_x,
           min(tile_row)          AS min_tile_y,
           max(tile_column)       AS max_tile_x,
           max(tile_row)          AS max_tile_y
    FROM tiles
    GROUP BY zoom_level"
        );
```